src/libraries/System.Formats.Tar/tests/TarWriter/TarWriter.WriteEntry.Tests.cs

```diff
@@ -299,5 +299,61 @@ public void WriteTimestampsBeyondOctalLimit(TarEntryFormat format)
                 }
             }
         }
+
+        [Theory]
+        [InlineData(TarEntryFormat.V7)]
+        // [InlineData(TarEntryFormat.Ustar)] https://github.com/dotnet/runtime/issues/75360
+        [InlineData(TarEntryFormat.Pax)]
+        [InlineData(TarEntryFormat.Gnu)]
+        public void WriteLongName(TarEntryFormat format)
+        {
+            string maxPathComponent = new string('a', 255);
+            WriteLongNameCore(format, maxPathComponent);
+
+            maxPathComponent = new string('a', 90) + new string('b', 165);
+            WriteLongNameCore(format, maxPathComponent);
+
+            maxPathComponent = new string('a', 165) + new string('b', 90);
+            WriteLongNameCore(format, maxPathComponent);
```

Comment:
Are these three cases actually testing anything different from each other? I'd suggest instead testing various lengths of interest, e.g. 99, 100, 101, 254, 255, 256.

Reply:
the last two cases were for testing that https://github.com/dotnet/runtime/issues/75360 didn't occur with other formats. ustar was flipping Name(100) and Prefix(155) order on read. I addressed your suggestion, using `Random` should be asymmetric enough to catch any name flipping.